com.woltlab.wcf/templates/headIncludeJavaScript.tpl

```diff
@@ -16,6 +16,7 @@
 	var ENABLE_PRODUCTION_DEBUG_MODE = {if ENABLE_PRODUCTION_DEBUG_MODE}true{else}false{/if};
 	var ENABLE_DEVELOPER_TOOLS = {if ENABLE_DEVELOPER_TOOLS}true{else}false{/if};
 	var WSC_API_VERSION = {@WSC_API_VERSION};
+	var PAGE_TITLE = '{PAGE_TITLE|language|encodeJS}';
```

Comment:
I really dislike all these global JavaScript variables, because they are not really ergonomic. Especially in TypeScript where they need to be defined in a `.d.ts`. Maybe this could be stored as a `data-` attribute on `<html>` in `documentHeader.tpl`? This one is also easy to access as `document.documentElement`.

Reply:
I thought about it, but placing it on `<html>` is a bit odd, because it may contain UTF-8. There is already the issue of the `<title>` being placed before the meta charset definition, adding another possible UTF-8 sequence before makes it even worse?

Reply:
Ah, indeed. Good point. Disregard my suggestion then. I don't have a suggestion for a better location then.